src/object.di

```diff
@@ -632,17 +632,17 @@ private void _destructRecurse(E, size_t n)(ref E[n] arr)
 
 private string _genFieldPostblit(S)()
 {
+    import core.internal.convert : itoa;
     import core.internal.traits : hasElaborateCopyConstructor;
-    import core.internal.traits : FieldNameTuple;
 
     string code;
-    foreach(fieldName; FieldNameTuple!S)
+    foreach(i, FieldType; typeof(S.init.tupleof))
     {
-        static if(hasElaborateCopyConstructor!(typeof(__traits(getMember, S.init, fieldName))))
+        static if(hasElaborateCopyConstructor!FieldType)
         {
             code ~= `
-                _postblitRecurse(s.` ~ fieldName ~ `);
-                scope(failure) _destructRecurse(s. ` ~ fieldName ~ `);
+                _postblitRecurse(s.tupleof[` ~ itoa(i) ~ `]);
```

Comment:
It appends a `u` for unsigned but apart from that it appears to work. If it can be relied on maybe we should use that instead.